generator/generate_service.go

```diff
@@ -1683,35 +1683,39 @@ func (g *generateCmd) generateRun() (*PartialGenerator, error) {
 			jen.Lit("URL"),
 			jen.Id("*zipkinURL"),
 		),
-		jen.List(jen.Id("collector"), jen.Err()).Op(":=").Qual(
-			"github.com/openzipkin/zipkin-go-opentracing", "NewHTTPCollector",
+		jen.Id("reporter").Op(":=").Qual(
+			"github.com/openzipkin/zipkin-go/reporter/http", "NewReporter",
 		).Call(jen.Id("*zipkinURL")),
+		jen.Defer().Id("reporter").Dot("Close").Call(),
+		jen.List(jen.Id("endpoint"), jen.Id("err")).Op(":=").Qual(
+			"github.com/openzipkin/zipkin-go", "NewEndpoint",
+		).Call(
+			jen.Lit(g.name),
+			jen.Lit("localhost:80"),
+		),
 		jen.If(jen.Err().Op("!=").Nil()).Block(
 			jen.Id("logger").Dot("Log").Call(
 				jen.Lit("err"),
 				jen.Id("err"),
 			),
 			jen.Qual("os", "Exit").Call(jen.Lit(1)),
 		),
-		jen.Defer().Id("collector").Dot("Close").Call(),
-		jen.Id("recorder").Op(":=").Qual(
-			"github.com/openzipkin/zipkin-go-opentracing", "NewRecorder",
-		).Call(
-			jen.Id("collector"),
-			jen.Lit(false),
-			jen.Lit("localhost:80"),
-			jen.Lit(g.name),
-		),
-		jen.List(jen.Id("tracer"), jen.Id("err")).Op("=").Qual(
-			"github.com/openzipkin/zipkin-go-opentracing", "NewTracer",
-		).Call(jen.Id("recorder")),
+		jen.Id("localEndpoint").Op(":=").Qual("github.com/openzipkin/zipkin-go", "WithLocalEndpoint").Call(jen.Id("endpoint")),
+		jen.List(jen.Id("nativeTracer"), jen.Id("err")).Op(":=").Qual(
+			"github.com/openzipkin/zipkin-go", "NewTracer",
+		).Call(jen.Id("reporter"), jen.Id("localEndpoint")),
 		jen.If(jen.Err().Op("!=").Nil()).Block(
 			jen.Id("logger").Dot("Log").Call(
 				jen.Lit("err"),
 				jen.Id("err"),
 			),
 			jen.Qual("os", "Exit").Call(jen.Lit(1)),
 		),
+		jen.Id("tracer").Op("=").Qual(
+			"github.com/openzipkin-contrib/zipkin-go-opentracing", "Wrap",
+		).Call(
+			jen.Id("nativeTracer"),
+		),
 	).Else().If(jen.Id("*lightstepToken").Op("!=").Lit("")).Block(
 		jen.Id("logger").Dot("Log").Call(
 			jen.Lit("tracer"),
```

Comment:
While we are changing this can we also update:
```
	defer lightsteptracergo.FlushLightStepTracer(tracer)
```
to
```
	defer lightsteptracergo.Flush(context.Background(), tracer)
```

`FlushLightStepTracer` is deprecated now.